lib/macho/load_commands.rb

```diff
@@ -268,7 +268,8 @@ def initialize(lc, lc_str)
         if view
           lc_str_abs = view.offset + lc_str
           lc_end = view.offset + lc.cmdsize - 1
-          @string = view.raw_data.slice(lc_str_abs..lc_end).delete("\x00")
+          raw_string = view.raw_data.slice(lc_str_abs..lc_end)
+          @string = raw_string.split("\x00").first
```

Comment:
Note that `String#partition` or `String#rpartition` is a more efficient alternative to `String#split` if we're only interested in the first/last delimited chunk. It's also less error-prone if we ever come across an empty string (valid, but consists entirely of null bytes). Try `"\x00\x00\x00".split("\x00")` to see what I mean.

In this case I would prefer a solution that makes sure the terminating `\x00` is always there, to make sure we're actually dealing with a null-terminated string. This could work (and was my first idea):

```ruby
null_index = raw_string.index("\x00")
raise "An appropriate exception" if null_index.nil?
@string = raw_string[0...null_index]
```

Using `String#partition` turns out to be more succinct (and still allows for validation):

```ruby
@string, null_byte, _ = raw_string.partition("\x00")
raise "An appropriate exception" if null_byte != "\x00"
```

Another simple alternative (that I'm less fond of because it doesn't validate):

```ruby
raw_string.unpack("Z*")
```

(Sorry for the potentially confusingly long comment. I prefer the `String#partition` variant, but I figured you might benefit from seeing all the other stuff that popped into my head while thinking about this.)